### ISSUE TITLE
mount module no swap support

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29647]
From @cherusk on 2016-09-16T16:58:59Z
Come back to me asap, please. Were prepared for impl. if OK.
##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

mount module
##### ANSIBLE VERSION

```
ansible 2.0.0.2
```

Copied from original issue: ansible/ansible-modules-core#4885

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (111/100 (111%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext8.iad.github.net ([192.30.252.199]:59002 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drStN-002Rvu-GE
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:53:05 -0500
Date: Mon, 11 Sep 2017 10:52:24 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152344;
	bh=86P1s5s4L4Nuh65kEt8xV8fhQIi5OOttulh6F40uxpE=;
	h=From:Reply-To:To:Cc:Subject:List-ID:List-Archive:List-Post:
	 List-Unsubscribe:From;
	b=mUEIU1LV9Ln9Z6R4ZS2X2KyfP9TeXPM0mD1wMR5jAbExjysqnaBzTbOT7QTcNhlcI
	 MSMNUEHVeUqe56qlV02H/aQ4wmXyBcyNZFXS1zHDSYJNSHT3hvVRntkNOty+2D9arz
	 akfCNgJhJLj8EiElrcgAqUfaADJ/xeauSlodhJwY=

**Comment by ansibot** [comment]
From @ansibot on 2016-09-16T16:58:59Z



@grossws ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2016-09-16T16:58:59Z

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @cherusk on 2016-09-16T16:58:59Z

no update ... so no response

**Comment by ansibot** [comment]
From @romain-dartigues on 2016-09-16T16:58:59Z

I'm also interested in a development of this feature.
I started to patch `mount.py` on my side, but I need community input for the design of the options... Does a PR related to this issue would help?

Related to ansible/ansible-modules-extras#3345

Regards.

**Comment by ansibot** [comment]
From @cherusk on 2016-09-16T16:58:59Z

**Got it working** in the meantime, though a little **awry**:

```
  - name: make swap space permanent
    mount: src={{ dev }} name=swap fstype=swap state=present opts=defaults passno=0 dump=0
```

**Comment by ansibot** [comment]
From @romain-dartigues on 2016-09-16T16:58:59Z

Nice, but it doesn't `swapon` it for you :^)
I would have liked a similar comportment than if it was a regular filesystem, especially to be able to treat swap and other FS in the same way in a `with_item` directive, for example.

**Comment by ansibot** [comment]
From @mscherer on 2016-09-16T16:58:59Z

yeah, I would assume doing the swapon would be coherent. I however do not know enough of the mount module to know if that's a good idea or not.

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (115/100 (114%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:51278 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSvD-002SyW-KF
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:54:59 -0500
Date: Mon, 11 Sep 2017 10:54:19 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152459;
	bh=wiT0Rel/EP8TgRxHRTyeYwUHZoCMF2PHOC4pYtJHi8A=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=XG5N/dfLwlSRXNaeRNUm3HGLWcB2jbXNmhQuYAd2kGQn/JFYE5+kDBZ+lI/UyVsHX
	 p5E2PBjwVMWORB3nYcTcpvmIGAQhruPFpBr+MIZtTKqjRK4TaIRGxfQyqkwWwMVtxT
	 tQVcTddApewTKM9mPrfd6xXcZYVPY/JggkWTWWW0=

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (117/100 (117%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:54570 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSvL-002T2V-VQ
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:55:08 -0500
Date: Mon, 11 Sep 2017 10:54:27 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152467;
	bh=qHbuBxz/QwZuKmF5irLqMr4T/12xuKKiQBdOPpnbVts=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=Q+4VgHbN9cRTYs29d4IPko2PJRr+vduTW77BROIoqwNvJohavyWgQWHB2WLBmhlL1
	 GuFEzGI4rqGIQMJ8y4pbaESZ16kRaFgrUkEj/liMG0/+8H4ys6V5xGiDst1MXlCMnN
	 DtcBACHA+F/cWjox9dBXUoL0jp4+b5HchAzsJPpE=

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (118/100 (118%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext8.iad.github.net ([192.30.252.199]:36270 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSvP-002T4a-Ml
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:55:11 -0500
Date: Mon, 11 Sep 2017 10:54:19 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152471;
	bh=I+x+fNbrYhC7cAEPgFUyhXBar7R6NOILRCmUhZryigk=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=lY0NAdiIjM3ATvWIQ3Q9wIWo5v1+w6o46NVL3NCRQPnHNKrn8zgZryx9tGr+hWfl3
	 0+eCSbCVnr2iB0jSIzCtQX9a0mHvR4eSmE5nlwlR+WyK6muyRo7feEsEMsR9P/efBK
	 PHe5+mIpKUyLHaGbvkLF0G7mvD0IOZte7zvMd+7I=

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (121/100 (121%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:38998 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSvX-002T8U-2M
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:55:19 -0500
Date: Mon, 11 Sep 2017 10:54:31 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152478;
	bh=Uw0P1BD8x+I/YdkW1Wknvyfs58ibY1bxiSgpXvOBEbg=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=ZyryW5Xd4bd3+MPH/vu5Yb1QkKqFtDAo/JeMgRWX40IkPl1i145L6mf9W8YgEHTW6
	 4RJkSIinUNs+Mt2VpVDCQOKsHeYlaaqI6fg7WYUtHDUw5VQ9Bwi4OymjAMqO5HEvNF
	 hlEFtcFgt1ufhgBxux+IOsUNgRox50B5FFAKce8s=

**Comment by ansibot** [comment]
cc @jtyr
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jtyr** [comment]
@cherusk Please could you provide more details what this issue is about? I did not get it from the issue description.

**Comment by romain-dartigues** [comment]
Duplicates #23632

**Comment by kees-closed** [comment]
2018, still can't mount swap with Ansible?

**Comment by ansibot** [comment]
cc @skvidal
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by romain-dartigues** [comment]
> 2018, still can't mount swap with Ansible?

Original PR to support swap, two years ago in 2016 via ansible/ansible-modules-extras#3345
If my code is bad, please say it so I can fix it...

**Comment by kees-closed** [comment]
@romain-dartigues have you seen #30090? Haven't tested that one for a while, I already changed my system configuration since it didn't work.

Also mounting swap doesn't work, using:
```
ansible 2.7.2
  config file = /home/kjong/git/ansible/ansible.cfg
  configured module search path = ['/home/kjong/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.1 (default, Nov 23 2018, 10:01:49) [GCC 8.2.1 20181105 (Red Hat 8.2.1-5)
```

This is the hostvar, which includes the data structure for the mount task:
```
mount:
  - name: /var/log
    src: LABEL=log
    fstype: ext4
    opts: defaults,noatime,noexec,nosuid,nofail
    passno: "2"
  - name: swap
    src: LABEL=swap
    fstype: swap
    opts: sw,nofail
```

Task:
```
  - block:
    - name: mount filesystem
      mount:
        name: '{{ item.name | mandatory }}'
        src: '{{ item.src | mandatory }}'
        fstype: '{{ item.fstype | default("auto") }}'
        opts: '{{ item.opts | default("defaults") }}'
        state: '{{ item.state | default("mounted")  }}'
        dump: '{{ item.dump | default("0") }}'
        passno: '{{ item.passno | default("0") }}'
        fstab: '{{ item.fstab | default("/etc/fstab") }}'
      loop: '{{ mount }}'
      when: mount is defined
      tags: mount
```

Which fails, unknown filesystem swap. All other filesystems I feed the task work without problems. It does create the /etc/fstab entry, but mounting it doesn't work. Have to do that manually.
```
PLAY [server] **************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************
Wednesday 28 November 2018  19:55:35 +0100 (0:00:00.124)       0:00:00.124 ****
ok: [paleobits]

TASK [common : include os environment vars] ********************************************************************************************
Wednesday 28 November 2018  19:55:37 +0100 (0:00:02.185)       0:00:02.309 ****
ok: [paleobits]

TASK [common : mount filesystem] *******************************************************************************************************
Wednesday 28 November 2018  19:55:37 +0100 (0:00:00.078)       0:00:02.388 ****
ok: [paleobits] => (item={'name': '/var/log', 'src': 'LABEL=log', 'fstype': 'ext4', 'opts': 'defaults,noatime,noexec,nosuid,nofail', 'passno': '2'})                                                                                                                             
failed: [paleobits] (item={'name': 'swap', 'src': 'LABEL=swap', 'fstype': 'swap', 'opts': 'sw,nofail'}) => {"changed": false, "item": {"fstype": "swap", "name": "swap", "opts": "sw,nofail", "src": "LABEL=swap"}, "msg": "Error mounting swap: mount: unknown filesystem type 'swap'\n"}

PLAY RECAP *****************************************************************************************************************************
paleobits                  : ok=2    changed=0    unreachable=0    failed=1   

Wednesday 28 November 2018  19:55:38 +0100 (0:00:01.442)       0:00:03.831 ****
===============================================================================
setup ------------------------------------------------------------------- 2.19s
common ------------------------------------------------------------------ 1.52s
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
total ------------------------------------------------------------------- 3.71s

```

**Comment by nre-ableton** [comment]
Is there any update on this issue? I am still running into this error:

Example task:

```yaml
- name: Ensure swapfile is mounted
  become: true
  mount:
    path: none
    src: "/swapfile"
    fstype: swap
    opts: "sw"
    state: mounted
```

Ansible output:

```
TASK [Ensure swapfile is mounted] ************************************************************
Thursday 05 December 2019  15:17:59 +0100 (0:00:00.052)       0:02:42.229 *****
fatal: [linux-bionic-staging]: FAILED! => {"changed": false, "msg": "Error mounting none: mount: /home/ci/none: unknown filesystem type 'swap'.\n"}
```

**Comment by RobVerduijn** [comment]
This is really weird
when using a playbook it fails, but an ad-hoc command succeeds
```
---
- name: mount lv swap
  hosts: test.example.com
  gather_facts: true

  tasks:

  - name: mount lv swap
    mount:
      path: none
      src: /dev/vg_data/lv_swap
      fstype: swap
      state: mounted
      opts: sw
      passno: '0'
      dump: '0'

PLAY [mount lv swap] *******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************************************************************************************
ok: [katello.lab.home]

TASK [mount lv swap] *******************************************************************************************************************************************************************************************************
fatal: [katello.lab.home]: FAILED! => {"changed": false, "msg": "Error mounting none: mount: unknown filesystem type 'swap'\n"}

PLAY RECAP *****************************************************************************************************************************************************************************************************************
test.example.com           : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0 
```

using an ad-hoc command it works 
```
ansible test.example.com -m mount -a 'path=none src=/dev/vg_data/lv_swap fstype=swap opts=sw state=present passno="0" dump="0"'
test.example.com | SUCCESS => {
    "changed": false,
    "dump": "0",
    "fstab": "/etc/fstab",
    "fstype": "swap",
    "name": "none",
    "opts": "sw",
    "passno": "0",
    "src": "/dev/vg_data/lv_swap"
}

```
using the latest ansible on fedora32
```
ansible 2.9.7
  config file = /home/rob/code/lab_home/ansible.cfg
  configured module search path = ['/home/rob/code/ansible-freeipa/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.2 (default, Feb 28 2020, 00:00:00) [GCC 10.0.1 20200216 (Red Hat 10.0.1-0.8)]
```

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/system/mount.py -> https://galaxy.ansible.com/ansible/posix

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
